crates/noirc_evaluator/src/ssa/optimizations.rs

```diff
@@ -233,7 +233,7 @@ fn cse_block_with_anchor(
 
                         new_list.push(*ins_id);
                     } else if let Some(similar) = anchor.find_similar_instruction(&operator) {
-                        debug_assert!(similar != ins.id);
+                        assert!(similar != ins.id);
```

Comment:
If we're asserting `!=`, we should use `assert_ne` instead since this will print out both operands if they differ

Reply:
Done